src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts

```diff
@@ -300,6 +302,14 @@ export class InlineChatWidget {
 		}
 	}
 
+	getCodeBlockInfo(): Promise<IReference<IResolvedTextEditorModel>> | undefined {
```

Comment:
Just hard-coding this to be the first (`0`) is fishy. I understand the terminal use-case but please make the index an argument and then it is up to terminal how to call this. Mid-term this needs to be tackled differently tho